### PR TITLE
PATCH maps/:mapID

### PR DESCRIPTION
**Pull request by laees**

### CODE REVIEW

File: server/src/modules/maps/maps.service.ts [hunk 1]
```diff
@@ -248,6 +248,39 @@ export class MapsService {
         return mapDB.id;
     }
 
+    async update(mapID: number, userID: number, updateBody: UpdateMapDto): Promise<void> {
+        const map = await this.mapRepo.get(mapID);
```
Comment (laees) [inline server/src/modules/maps/maps.service.ts:252]: I don't know if there are status flags that not every user is allowed to set. If there are then a check needs to be added here. 
As far as I could see the old API doesn't check for it so I didn't either

File: server/src/modules/maps/maps.controller.ts [hunk 2]
```diff
@@ -105,6 +105,27 @@ export class MapsController {
         return this.mapsService.get(mapID, userID, query.expand);
     }
 
+    @Patch('/:mapID')
+    @HttpCode(HttpStatus.NO_CONTENT)
+    @Roles(RolesEnum.MAPPER)
+    @ApiOperation({ summary: 'Updates a single maps status flag' })
```
Comment (tsa96) [inline server/src/modules/maps/maps.controller.ts:111]: Map's (use double quotes for whole string)

File: server/src/modules/maps/maps.service.ts (hunk 1 above)
Comment (tsa96) [reply server/src/modules/maps/maps.service.ts:252]: Yeah the permissions system here does seem a bit odd, let's just implement it the same way for now.

Though one thing you're missing, the old API is changing that the map is not in the `REJECTED` or `REMOVED` state, let's at least add that. So just return a 400 if it's in either of those states.

**Review by tsa96 — CHANGES_REQUESTED** [review]
Awesome stuff, very few issues. You handled the activities creation and testing very succinctly, so much nicer than old API!

**Review by tsa96 — APPROVED** [review]

**Pull request merged**